grammar.js

```diff
@@ -656,9 +656,23 @@ module.exports = grammar({
       seq("'", repeat(choice(/[^\\'\n]/, /\\./)), "'")
     )),
 
-    template_string: $ => token(seq(
-      '`', repeat(/[^`]/), '`'
-    )),
+    template_string: $ => seq(
+      '`',
+      repeat(choice(
+        $.template_chars,
+        $.template_substitution
+      )),
+      '`'),
+
+      template_chars: $ => prec.right(repeat1(
```

Comment:
Can you try putting a `token` around this? I think that currently it will create a node for each character.